app.js

```diff
@@ -0,0 +1,84 @@
+var path = require('path');
+var fs = require('fs');
+var express = require('express');
+var Router = require('react-router');
+
+var indexStaticWatcher = require('./lib/index-static-watcher').create();
+var PORT = process.env.PORT || 8008;
+var PRODUCTION = (process.env.NODE_ENV === 'production');
+var DIST_DIR = path.join(__dirname, 'dist');
+
+var indexStatic;
+var router;
+var app = express();
+
+var startProdApp = function() {
+  console.log([
+    'Production mode enabled. Note that "npm install" is assumed to',
+    'have recently been run with NODE_ENV="production". If this is not',
+    'the case, some or all static assets may be out of date.'
+  ].join('\n'));
+  indexStaticWatcher.build(function(err, newIndexStatic) {
+    if (err) {
+      throw err;
+    }
+
+    console.log('Built server-side bundle.');
```

Comment:
> Where are the .html files the dynamic server is serving then (or how did the HTML get generated from `.jsx`)?  :sob: :sob: :sob:

Oh, that's actually what puts the "dynamic" in "dynamic server"--it's running the HTML renderer for each request and sending the generated response to the client without ever storing it in the file system (since that would just slow things down). That's basically what's happening in lines 45-64 of `app.js`, when the server calls `indexStatic.generate()`.

To put it another way: the static site uses `indexStatic.generate()` to generate every unique page for the site once and writes each page out to a HTML file in the `dist` directory, whereas the dynamic server uses that same function to dynamically generate the page being requested on-the-fly and send it back to the user without writing it to the file system.

In a lot of ways the static site approach seems less error-prone and more performant because we generate the whole site at deployment time, and S3/CloudFront takes care of scaling it. However, due to the reasons given in #585, we want to be able to migrate to the dynamically-rendered approach if needed (or even perhaps combine the two approaches somehow).